scripts/lib/node_cache.py

```diff
@@ -9,20 +9,31 @@ def setup_node_modules(production: bool = DEFAULT_PRODUCTION) -> None:
     if os.path.islink("node_modules"):
         os.unlink("node_modules")
 
+    skip = False
+
     try:
         with open("node_modules/.pnpm/lock.yaml") as a, open("pnpm-lock.yaml") as b:
             if a.read() == b.read():
-                return
+                skip = True
     except FileNotFoundError:
         pass
 
-    run(
-        [
-            "/usr/local/bin/corepack",
-            "pnpm",
-            "install",
-            "--frozen-lockfile",
-            "--prefer-offline",
-            *(["--prod"] if production else []),
-        ]
-    )
+    # We need this check when switching between branches without `help-beta`
+    # package. `node_modules` will be removed when working on a non `help-beta`
+    # branch, but if `node_modules/.pnpm/lock.yaml` has not been updated by that
+    # branch, we will end up in a situation where we might not have `node_modules`
+    # even when we run the provision command.
+    if not os.path.exists("help-beta/node_modules"):
+        skip = False
```

Comment:
That situation isn’t possible. `pnpm-lock.yaml` now references `help-beta` in the `importers` section, so switching from a branch without it will by definition update `pnpm-lock.yaml`.

Reply:
This was happening to me in cases where I had a `run-dev` process running in background and I switched branches. In that case, I didn't have to run provision on the other branch, but sometimes I cleared the extra node_modules folder with `git clean -fd` when in the other branch. Switching back to help center PR in that case would not run pnpm install.

Not sure if we should have the special code for this case or not, since, to encounter this error, I needed to skip running provision on the other branch